custom_components/mikrotik_router/mikrotik_controller.py

```diff
@@ -634,3 +665,229 @@ def get_queue(self):
             upload_burst_time, download_burst_time = self.data["queue"][uid]["burst-time"].split('/')
             self.data["queue"][uid]["upload-burst-time"] = upload_burst_time
             self.data["queue"][uid]["download-burst-time"] = download_burst_time
+
+    # ---------------------------
+    #   get_dhcp
+    # ---------------------------
+    def get_dhcp(self):
+        """Get DHCP data from Mikrotik"""
+
+        self.data["dhcp-server"] = parse_api(
+            data=self.data["dhcp-server"],
+            source=self.api.path("/ip/dhcp-server"),
+            key="name",
+            vals=[
+                {"name": "name"},
+                {"name": "interface", "default": ""},
+            ]
+        )
+
+        # Build list of local DHCP networks
+        dhcp_networks = parse_api(
+            data={},
+            source=self.api.path("/ip/dhcp-server/network"),
+            key="address",
+            vals=[
+                {"name": "address"},
+            ],
+            ensure_vals=[
+                {"name": "address"},
+            ]
+        )
+        self.local_dhcp_networks = [IPv4Network(network) for network in dhcp_networks]
```

Comment:
it would be probably better to update self.data["dhcp_networks"] within get_dhcp_networks directly instead of adding new variable.
we cant change key, but adding new field should be easy. like:
            ensure_vals=[
                {"name": "IPv4"},
            ]

Reply:
key "IPv4Network" is available as self.data["dhcp-network"]